src/openrct2/entity/Guest.cpp

```diff
@@ -5444,6 +5446,44 @@ void Guest::UpdateWalking()
     }
 }
 
+void Guest::UpdateWaitingAtCrossing()
+{
+    if (!IsActionInterruptable())
+    {
+        UpdateAction();
+        Invalidate();
+        if (!IsActionWalking())
+            return;
+    }
+
+    Action = PeepActionType::Idle;
+    NextActionSpriteType = PeepActionSpriteType::WatchRide;
+    SwitchNextActionSpriteType();
+
+    if (HasFoodOrDrink())
+    {
+        if ((scenario_rand() & 0xFFFF) <= 1310)
+        {
+            Action = PeepActionType::EatFood;
+            ActionFrame = 0;
+            ActionSpriteImageOffset = 0;
+        }
+
+        UpdateCurrentActionSpriteType();
+
+        return;
```

Comment:
Just checking: guests with food or drinks will never perform any other animation, is that intended?

Reply:
Yes, it's intended. If guests with food or drinks would start waving (for example), it looks weird since the food or drink would suddenly disappear from the sprite and return once the animation has been completed.

Please note that other animations can be triggered by regular animation logic outside this code block. For example, I have seen guests looking at their map whilst waiting at the crossing for a passing train. The call `!IsActionInterruptable()` followed by `UpdateAction()` ensures these animations will now play too, which didn't happen before.